regression-tests.recursor-dnssec/test_LargeAnswer.py

```diff
@@ -81,12 +99,20 @@ def datagramReceived(self, datagram, address):
 
         if request.question[0].name == dns.name.from_text('AAAA.large-answer.example.'):
             value = 'A'
-        else:
+            final_count = 5
```

Comment:
This makes the large answer test useless, as it can no longer catch an of-by-one if the answer from an auth matches exactly `udp-truncation-threshold`. I'm guessing you altered that test because the recursor now always keep space to add the `EDNS OPT` record when the client supports `EDNS0`, so the existing answer was truncated. Please consider simply disabling `EDNS` in the queries sent in `testLargeAnswer()` instead, if I'm not mistaken it should be enough to restore the existing behavior.

Reply:
Disabling EDNS from the client would make the nameserver send a TC response, as the 1680 bytes won't fit into the reply packet. I'll make a noEDNS test that will max out the packet size.

Reply:
But it did fit before without EDNS, what am I missing?

Reply:
the tests always had EDNS enabled, the client would send 4096 as the payload size, the recursor would send 1680 to the "auth", get a response of exactly that size and stuff in the packet to the client and forget about the EDNS OPT record in the reply packet as it would not fit.

Reply:
We discussed this on IRC, because we now reserve space for the OPT, truncation works a little different. Added a test without EDNS to also test 512 byte truncation.

Reply:
But does the modified version test the issue the initial version was testing for, an off-by-one in the buffer processing response from authoritative servers whose size is exactly the same than our outgoing buffer size? The fact that `_udpTruncationThreshold` did not change while the size of the answer generated from `UDPLargeResponder()` did makes me think it doesn't.

Reply:
aha, *now* I understand what we were testing. This had nothing to do with the response to the client but with the response from the auth and the outgoing buffer size